evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -2104,3 +2094,95 @@ def test_component_graph_compute_final_component_features_with_sampler(
     assert len(mock_estimator_fit.call_args[0][0]) == expected_length
     features_for_estimator = component_graph.compute_final_component_features(X, y)
     assert len(features_for_estimator) == len(y)
+
+
+@patch("evalml.pipelines.components.Imputer.transform")
+@patch("evalml.pipelines.components.OneHotEncoder.transform")
+def test_component_graph_transform(
```

Comment:
What exactly is this testing?  The components are all mocked out to just pass through X.  Wouldn't it actual demonstrate that the new `component_graph.transform()` is working by actually having it do the transformation(s)?

Reply:
My philosophy here was that it doesn't matter what the components individual `transform` methods return, hence why I'm mocking each of the transform. The component graph should just be responsible for and tested for being able to string those along... but maybe there is a danger in keeping the original value, X, though, rather than returning some dummy value. I can change that if it makes it more clear!

Reply:
Yup!